back/lib/cohere_multilingual_embeddings.rb

```diff
@@ -0,0 +1,46 @@
+class CohereMultilingualEmbeddings
+  def initialize(**params)
+    if !params.key? :region # 'eu-central-1'
+      raise 'No AWS region specified for Anthropic Claude model.'
+    end
+
+    @client = Aws::BedrockRuntime::Client.new(params)
+  end
+
+  def self.token_count(str)
+    # From https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-embed.html:
+    # "For optimal performance, we recommend reducing the length of each text to less than 512 tokens. 1 token is about 4 characters."
+    (str.size / 4.0).ceil
+  end
+
+  def embedding(text, **params)
+    resp = @client.invoke_model invoke_params(text, **params)
+    body_completion resp.body.string
+  end
+
+  protected
+
+  def model_id
+    'cohere.embed-multilingual-v3'
+  end
+
+  def input_type
+    'search_document' # Alternatively 'search_query'
```

Comment:
From the docs, it's not entirely clear which `input_type` we should use. I think that `search_query` might be recommended when the user types a search term, while `search_document` might be recommended to find documents similar to a given document?